### PR TITLE
Clear all categories

### PR DESCRIPTION
**Pull request by FMA126**
## Description
Refactor of categories feature

Fixes #
It looks better, the categories clear, but it is laggy.  Also, the previous way way of changing the state was incorrect because it mutated state.  My proposed pr is changing the state correctly with setState()

## Motivation and Context
After 4 weeks I finally got it to work

## Type of change

- [x ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change (updates to README and related files)

### CODE REVIEW

**Comment by johnpyp** [comment]
Functionality looks good to me, and it isn't laggy like you say for me (even throttling cpu). However, it doesn't currently follow the "styleguide" / eslint requirements. To check for that, make sure you have the latest version of node_modules, and do `yarn lint`. That should tell you where your errors are. 

Most of the them are autofixable, with `yarn lint --fix`, but a few of them aren't, most notably adding propTypes and removing unused variables.

**Comment by FMA126** [comment]
Awesome, thanks John Paul. Just pushed to the branch again, should be error free on the linter side.

File: src/components/AdminPage/CategoryList.js [hunk 1]
```diff
@@ -2,67 +2,113 @@ import React, { Component } from "react";
...
+      </ListGroupItem>
     ));
 
     return (
-      <Form>
+      <div>
         <Label>Filter by Category</Label>
-        {categoryMenuItems}
-      </Form>
+        <Button color="info" className="w-100" onClick={this.clearChecks}>
```
Comment (RajChaudhry) [inline src/components/AdminPage/CategoryList.js]: Having this render below the categories makes a bit more sense on the UX perspective

**Comment by RajChaudhry** [comment]
> Cool, looks good to me.
> 
> Edit: Sorry for the weird review history here, I accidentally approved it, but I don't think I'm supposed to have the power to review & merge.

@johnpyp I encourage everyone to review PR's. In my opinion, the more eyes on something, the better. However, it'd be better if one of the more senior members were to merge.

**Pull request merged**